### ISSUE TITLE
Add support Vala for linux 

### ISSUE THREAD

**Issue by listepo** [flutter/flutter#110410]
It would be very nice to be able to create apps and plugins for linux with [Vala](https://vala.dev/). C/C++ is very good, but the entry threshold is very high.

**Comment by stuartmorgan** [comment]
/cc @robert-ancell for input (IIRC we discussed this briefly during the initial bringup?), and @cbracken.

There are a few elements of support we could consider if we wanted to do this:
- Having `doctor` check for the necessary tooling (is it just `valac`, or are there other components?). This would required if we want to allow plugins to be written with Vala without causing a lot of problems.
- Providing bindings for the Flutter embedding APIs.
- Templates and template language switches for apps and plugins.

Some important considerations here:
- How popular is Vala? Having a safer option than C would be nice, but is Vala the best choice? In general we only support a second language when there's a strong signal that it's the successor language for that platform; are there stats or other things that support that conclusion for Vala in the context of GTK applications?
- How stable is the language? I see that it's pre-1.0; how much churn is there in compatibility? For apps that's not a big deal, but if plugins are written in a high-churn language it will really hurt the ecosystem over time.
- How easy is getting the necessary tooling? The docs say "Although Ubuntu is based on Debian, the packages in the Ubuntu repositories are often out of date, especially the LTS releases. To install the most recent version of Vala you should add the [Vala Team PPA](https://launchpad.net/~vala-team/+archive/ubuntu/ppa) to your software sources". Making people add apt repos just to install Flutter isn't ideal.

**Comment by robert-ancell** [comment]
My first question would be what sort of apps and plugins did you have in mind? This is only going to be useful if a significant part of the application or plugins is to be written in native code as opposed to Dart/Flutter. In my experience I've almost never used plugins with Flutter preferring to use Dart and falling back to Dart+FFI if it that isn't possible.

Changing the application from C to Vala should be do-able right now, this just requires rewriting the template in Vala as all the exposed Flutter APIs should be wrappable with a .vapi. This could be done with a separate github project that is then copied into projects that want to use it. This could be assessed for popularity to decide if this would warrant becoming an official template option in Flutter.

Writing plugins in Vala should also be do-able if they are pre-compiled to C for distribution. Again, if significant plugins made use of this then the next step would be to consider if Flutter should natively support valac and remove the pre-compile stage.

**Comment by robert-ancell** [comment]
I have used a lot of Vala in the past and it's a language I do love - much better than fighting C. Dart reminds me a lot of Vala particularly in writing a lot of asynchronous code.

**Comment by stuartmorgan** [comment]
> Writing plugins in Vala should also be do-able if they are pre-compiled to C for distribution.

Interesting, so it's straightforward to do just the transpiling step and stop with the C output?

If someone in the community wants to write up a step-by-step guide to writing a plugin that way and then link to it from here, that would be a good starting point to gauge interest.

**Comment by robert-ancell** [comment]
> Interesting, so it's straightforward to do just the transpiling step and stop with the C output?

Yes, `valac --ccode` will just spit out a .c file for the .vala input and can be compiled just like any other C code.

**Comment by listepo** [comment]
@stuartmorgan-g @robert-ancell Hi, if I find time to create a high-quality example, will it be considered for integration into Flutter?

**Comment by stuartmorgan-g** [comment]
Given that we haven't seen evidence of interest in this in the intervening years, it's not clear that this would rise to the level of something that would be incorporated into our official docs (especially since we'll be transitioning to emphasize Pigeon more in the docs, and that may add complexity to demonstrating using Vala for implementation).

As [noted above](https://github.com/flutter/flutter/issues/110410#issuecomment-1231008652), I think the next step would be for a community member interested in this to publish a guide elsewhere and link to it from here. Then it could be evaluated by developer relations, and either way would be discoverable for anyone finding this issue.
